File: leetcode/python/11.py
## Leetcode 11 - Container with Most Water
## Nov. 29, 2023

from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans, l, r = 0, 0, len(height) - 1
        while l < r:
            area = min(height[l], height[r]) * (r - l)
            ans = max(ans, area)
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return ans

File: leetcode/c++/11.cpp
// Leetcode 11 - Container with Most Water
// Nov. 29, 2023

#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0 , l = 0, r = height.size() - 1;
        while (l < r) {
            int area = min(height[l], height[r]) * (r - l);
            ans = max(ans, area);
            if (height[l] < height[r]) l++;
            else r--;
        }
        return ans;
    }
};